### ISSUE TITLE
don't know what I should add to support the `new` type

### ISSUE THREAD

**Issue by NormanPerrin** [gajus/eslint-plugin-jsdoc#92]
### Question
The linter throws an error of type `new` unknow.
How could I config the eslint to support this type?

### type reference
[function-new-type](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#function-new-type)

### error
```bash
Syntax error in type: {new(source: {documents: Map<string, D>}): DListConstructor} jsdoc/valid-types
```

### .eslintrc config
```json
{
    "plugins": [
        "jsdoc",
        "closure"
    ],
    "env": {
        "es6": true
    },
    "rules": {
        "jsdoc/check-param-names": 1,
        "jsdoc/check-tag-names": 1,
        "jsdoc/check-types": 1,
        "jsdoc/newline-after-description": 1,
        "jsdoc/require-description-complete-sentence": 1,
        "jsdoc/require-param": 1,
        "jsdoc/require-param-type": 1,
        "jsdoc/require-returns-type": 1,
        "jsdoc/valid-types": 1
    },
    "settings": {
        "jsdoc": {
            "additionalTagNames": {
                "customTags": ["template"]
            }
        }
    }
}
```

**Comment by gajus** [comment]
Whats the code?

**Comment by NormanPerrin** [comment]
```
/**
 * @param {{new(source: {documents: Map<string, D>}): DListConstructor}} ObjectiveClass
 */
```

**Comment by gajus** [comment]
Could you reference JSDoc manual that identifies this as a valid type declaration?

**Comment by bary12** [comment]
Note that your usage is not the same in the reference you linked
```js
function(new:goog.ui.Menu, string)
```

**Comment by NormanPerrin** [comment]
The suggestions I'm getting are ok, so the type should be fine.

It's not an issue of jsdoc types, as it's a google closure compiler type, so it's ok that it throws me that error. But I thought someone might know how to config that type to be a valid one.

**Comment by brettz9** [comment]
@NormanPerrin , you can file a PR at https://github.com/jsdoctypeparser/jsdoctypeparser/ which is the library we are using for parsing jsdoc types.

I'll close this out here, but you can follow https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50 which is tracking features from TypeScript, including "Construct signature" that appears to include the feature you have requested (and if it is specific to GCC, feel free to file a separate issue).
